compiler-rt/test/ubsan/TestCases/TypeCheck/vptr.cpp

```diff
@@ -23,7 +23,9 @@
 // RUN: %env_ubsan_opts=halt_on_error=1 not %run %t nN 2>&1 | FileCheck %s --check-prefix=CHECK-NULL-MEMFUN --strict-whitespace
 // RUN: %env_ubsan_opts=print_stacktrace=1 %run %t dT 2>&1 | FileCheck %s --check-prefix=CHECK-DYNAMIC --allow-unused-prefixes --check-prefix=CHECK-%os-DYNAMIC --strict-whitespace
 
-// RUN: (echo "vptr_check:S"; echo "vptr_check:T"; echo "vptr_check:U") > %t.supp
+// RUN: echo "vptr_check:S" > %t.supp
+// RUN: echo "vptr_check:T" >> %t.supp
+// RUN: echo "vptr_check:U" >> %t.supp
```

Comment:
Couldn't you rewrite the `echo` command w/ newlines to avoid the need for multiple RUN lines?

Alternatively, you could use use split-file instead. It will take some rewriting of the test, since you'll need to make the directory at the top of the file w/ `%t`, and then update uses of `%t` and %s`.

Reply:
I think having split files just makes the test harder to read.

I don't see anything really wrong with the three `RUN` lines.

Reply:
split-file is probably overkill here, but its a pattern we use elsewhere. If we can avoid 2 subshell invocations w/o losing much (if any) readability, we probably should try to do that, IMO. It isn't a huge performance win, but I don't find it much more readable split onto 3 lines w/ file redirection than spelled out as a single `echo`, and we may as well be a little bit more efficient.

Reply:
Sure, either way (except split-file) is fine with me.

Reply:
I used `echo -e` in a single `RUN` line to create the suppression file, which simplifies the process and avoids the need for multiple subshell invocations. This keeps the script efficient and easy to maintain.  Additionally, this approach is better because it minimizes changes to the file, unlike when using `split-file`. What are your thoughts on this approach?

Reply:
LGTM, but I will note that POSIX actually says `-e` is implementation dependent and that the default should be to interpret the escape characters: https://pubs.opengroup.org/onlinepubs/9799919799/. It appears that `/bin/echo` or bash's builtin do not follow this though :(

Reply:
IIRC we have other tests that use `\n` w/o `-e`, but I'm fine either way.

Reply:
I initially tried to use `\n` without `-e`, but I encountered an error. Given that, I decided to go with `echo -e`, which worked fine in this case.

Reply:
Yeah, I played around a bit and on my Linux `/bin/echo`  needed the `-e` to support the escape sequences. So did bash. zsh didn't.

POSIX is silent on that: "If the first operand consists of a '-' followed by one or more characters from the set {'e', 'E', 'n'}, or if any of the operands contain a <backslash> character, the results are implementation-defined." https://pubs.opengroup.org/onlinepubs/9799919799/utilities/echo.html

Reply:
Thanks for the explanation—that makes sense now. Given the inconsistencies, it seems like `echo -e` is the best option. Is it okay if I stick with that?

Reply:
I think so. Also, all matches for `; RUN: echo .*\\n` seem to use `-e`.

Reply:
Yes, I just checked, and all the matches are indeed using `-e`. Thank you for the feedback and guidance.